dataset/bucket.cpp

```diff
@@ -267,8 +296,9 @@ Variable histogram(const VariableConstView &data,
     spans = VariableConstView(merged);
   }
   return variable::transform_subspan<std::tuple<
-      args<double, double, double, double>, args<double, float, double, double>,
-      args<double, float, double, float>, args<double, double, float, double>>>(
+      args<float, double, float, double>, args<double, double, double, double>,
```

Comment:
Not really, especially since here we may want to limit the combinations, e.g., to require that edges and event coord values have the same type, to avoid compiling too many cases.